### DISCUSSION TITLE
trouble with the md:myst output

### DISCUSSION THREAD

**Discussion by parmentelat** [executablebooks/meta#295] [Q&A]
Hi folks

I'm currently running into an issue, which has to do with the md:myst output format 

I had filed an issue in jupytext at first, here
https://github.com/mwouts/jupytext/issues/759

digging a little further - i.e. trying to reproduce the issue within the jupytext test context - did not pan out

I could read here
https://myst-nb.readthedocs.io/en/latest/use/markdown.html?highlight=code-cell%20ipython3#syntax-for-code-cells
that the `ipython3` addition to `code-cell`  was optional, and I'm curious to know if there are any environment/settings that would cause the myst output to come with that, as all my attempts so far with the jupytext cli would not include that  `ipython3`, while otoh all my attempts at producing a myst from jupyter classic do include that extra `ipython3`

I'd be really interested if someone from the myst team could take a quick look at that issue
and let me know if I'm not simply missing something obvious here
or redirect me to the right people/channel if relevant

thanks a bunch !

**Comment by choldgraf** [comment]
Hey 👋 could you clarify your problem a bit? It is a bit hard to understand what is the specific thing you’re expecting to happen, and what is not happening.

**Comment by parmentelat** [comment]
sure :)

all my notebooks tagged as being 'md:myst' behave the same:
I expect to obtain the same textual representation whether I save under jupyter or if I use jupytext --to md:myst

my problem is I don't get 2 different outputs:
* from jupyter, all my code cells are fenced with

      ```{code-cell} ipython3
      # the python contents
      ```
* while from jupytext I more simply get 

      ```{code-cell}
      # the python contents
      ```

I was not able to reproduce the issue from the jupytext test env, but from there I always get the second form

so I am trying to figure what it is in my environment that causes my installed jupyter to end up producing the first form

hope this clarifies the matter - thanks  !

**Reply by choldgraf** [reply]
Ah - indeed that is strange, because the jupyter interface is just using jupytext under the hood to do the conversion, I think. I don't believe there is any "native" support for myst nb in jupyter

**Comment by chrisjsewell** [comment]
Yes it's optional and does not affect the execution in any way, it is used for syntax highlighting and is populated from the pymgments/lever key in the the notebook metadata

**Comment by parmentelat** [comment]
> Yes it's optional and does not affect the execution in any way, 

right, but the reason it is an impediment is because in some use cases I end up with a lot of meaningless diffs - which was the reason for going text in the first place ;-)

> it is used for syntax highlighting and is populated from the pymgments/lever key in the the notebook metadata

ah, this is interesting !
can you please refer to the code that does that ? 
should that metadata not impact both workflows ? 

thanks !

**Reply by choldgraf** [reply]
Could you describe more how you're creating the myst-notebook file from Jupyter? Are you just using jupytext's auto-converter extension?

**Comment by parmentelat** [comment]
in this particular instance, I had a bunch of ipynb files that I mass-converted to myst using  
`juytext --to md:myst` 
those files contained the untagged 

    ```{code-cell}

from that point, as soon as I load any of these files under jupyter classic, and save it from there, the `ipython3` tag gets added, even if no other change is made in the notebook

I am suspecting some odd configuration in my jupyter custom settings maybe, but before I scrub that thoroughly (which honestly feels scary) I'd really like to figure what on earth can be happening here :)

thanks !
